### DISCUSSION TITLE
NFT can’t be added as the ownership details do not match. Make sure you have entered correct information

### DISCUSSION THREAD

**Discussion by ProKing13** [Cyfrin/foundry-full-course-cu#1670] [Q&A]
I am having the below problem as all the other things works correctly means minting and deploying works properly but i can't PUG NFT in my wallet NFT part and the below error occurs:
```error
NFT can’t be added as the ownership details do not match. Make sure you have entered correct information
```
I have used the correct address and i have used the token id (0) but it gives me the above error.

**Comment by EngrPips** [comment]
Hello @MohammadArif13, Here is a discussion with a similar issue: https://github.com/Cyfrin/foundry-full-course-f23/discussions/1015#discussion-5738417. Please look and see if the suggested solutions would solve your issue.

**Reply by ProKing13** [reply]
@EngrPips but bro how to know my token id. I have searched a lot.

**Reply by EngrPips** [reply]
Hey @MohammadArif13, Are you currently building the `BasicNFT` or the `MoodNFT`?

**Reply by ProKing13** [reply]
@EngrPips Bro I  have build BasicNFT and i have deployed it correctly but when i want to import NFT to MetaMask account then the above error occurs.

**Reply by EngrPips** [reply]
Alright, How many times have you minted ans do you have your Nft contract address ?

**Reply by ProKing13** [reply]
this is Ineraction.s.sol file:
```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;
import {Script,console} from "forge-std/Script.sol";
import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";
import {BasicNft} from "../src/BasicNft.sol";

contract MintBasicNft is Script {
    string public constant PUG =
        "ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json";
    function run() external {
        address mostRecntlyDeployed = DevOpsTools.get_most_recent_deployment(
            "BasicNft",
            block.chainid
        );
        console.log("Below is rescekjddl;");
        console.log(mostRecntlyDeployed);
        mintNftOnContract(mostRecntlyDeployed);
    }
    function mintNftOnContract(address constractAddress) public {
        vm.startBroadcast();
        BasicNft(constractAddress).mintNft(PUG);
        vm.stopBroadcast();
    }
}
```


this is my BasicNft contract:
```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;
import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";

contract BasicNft is ERC721 {
    uint256 private s_tokeCounter;
    mapping (uint256 => string) private s_tokeIdToUri;
    constructor() ERC721("Dogie", "DOG") {
        s_tokeCounter = 0;
    }
    function mintNft(string memory tokeUri) public {
        s_tokeIdToUri[s_tokeCounter]=tokeUri;
        _safeMint(msg.sender,s_tokeCounter);
        s_tokeCounter++;
    }
    function tokenURI(
        uint256 tokenId
    ) public view override returns (string memory) {
        return s_tokeIdToUri[tokenId];
    }
}
```

the below is my DeployBasicNft:
```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;
import {Script,console} from "forge-std/Script.sol";
import {BasicNft} from "../src/BasicNft.sol";
contract DeployBasicNft is Script{
    function run() external returns(BasicNft){

        vm.startBroadcast();
        BasicNft basicNft=new BasicNft();
        vm.stopBroadcast();
        return(basicNft);
    }

}
```

**Reply by ProKing13** [reply]
@EngrPips Bro this the address:
```
0x027461043C61D325ddab894A5D750eAED496EFf8
```

**Reply by EngrPips** [reply]
Alright, can you maybe show your metamask interface where you are trying to add the NFT ?

**Reply by EngrPips** [reply]
Below is the `NFT` interface on my end 

<img width="1440" alt="Screenshot 2024-05-03 at 18 51 36" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/c23ba2cc-e872-4f28-8429-8f7bd7fe6648">

**Reply by ProKing13** [reply]
OK bro now I will share the screenshot with you


![Uploading MetaMask - Brave 5_4_2024 9_35_44 AM.png…]()

**Reply by ProKing13** [reply]
OK bro now I will share the screenshot with you


![Uploading MetaMask - Brave 5_4_2024 9_35_44 AM.png…]()

**Reply by ProKing13** [reply]
![Uploading MetaMask - Brave 5_4_2024 9_35_44 AM.jpg…]()

**Reply by ProKing13** [reply]
@EngrPips  bro sorry for sending it mulitiple time because my internet is so slow.
![MetaMask - Brave 5_4_2024 9_35_44 AM](https://github.com/Cyfrin/foundry-full-course-f23/assets/124627824/34a65e88-09c0-4b0f-8d35-5ba05feb7d6d)

**Reply by EngrPips** [reply]
Please share me the contract address of the Nft.

**Reply by ProKing13** [reply]
@EngrPips  bro i didn't get the point.
which contract address i should share with you?

Bro this is the address DeployBasicNft:
```
0x027461043C61D325ddab894A5D750eAED496EFf8
```

**Reply by EngrPips** [reply]
Thanks for sharing the address. The contract shows that you haven't minted any `Nft` yet. The only transaction on the contract address is the contract creation transaction.

<img width="1440" alt="Screenshot 2024-05-05 at 07 35 53" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/5b378a7b-119f-456d-92fb-6cf7c60a1c3e">

**Reply by EngrPips** [reply]
Please call the `Mint` function on the contract address either through cast or through a script. Make sure you use the private key of your sepolia wallet address. After you have minted successfully, you will be able to add the `Nft` to your metmask.

**Reply by ProKing13** [reply]
@EngrPips  bro so sorry disturbing You. I am new in solidity and now i want to learn. 

bro, I have mintNft function in BasicNft.sol  so please tell me how to call mintNft function.
all of my code is in above commants

BasicNft.sol

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;
import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";

contract BasicNft is ERC721 {
    uint256 private s_tokeCounter;
    mapping (uint256 => string) private s_tokeIdToUri;
    constructor() ERC721("Dogie", "DOG") {
        s_tokeCounter = 0;
    }
    function mintNft(string memory tokeUri) public {
        s_tokeIdToUri[s_tokeCounter]=tokeUri;
        _safeMint(msg.sender,s_tokeCounter);
        s_tokeCounter++;
    }
    function tokenURI(
        uint256 tokenId
    ) public view override returns (string memory) {
        return s_tokeIdToUri[tokenId];
    }
}
```

DeployBasicNft

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;
import {Script,console} from "forge-std/Script.sol";
import {BasicNft} from "../src/BasicNft.sol";
contract DeployBasicNft is Script{
    function run() external returns(BasicNft){

        vm.startBroadcast();
        BasicNft basicNft=new BasicNft();
        vm.stopBroadcast();
        return(basicNft);
    }

}
```

**Comment by EngrPips** [comment]
Hello @MohammadArif13, Are you taking Patrick's `Foundry FULL COURSE` on Cyfrin updraft? Here is the link https://updraft.cyfrin.io/courses/advanced-foundry/how-to-create-an-NFT-collection/introduction-to-nfts?lesson_format=video  to the `NFT` section of Patrick's `Foundry FULL COURSE,` He teaches everything in this section. Please let me know if you need any further assistance

**Reply by ProKing13** [reply]
@EngrPips Yes bro I am following Patrick's Foudry Full Course which have three parts. Now I am at the end of second part.

The same like I have done : https://updraft.cyfrin.io/courses/advanced-foundry/how-to-create-an-NFT-collection/deploy-nfts-on-testnet?lesson_format=video

**Reply by EngrPips** [reply]
Awesome. You could rewatch any lesson you don't fully understand.
